test/integration/hat-threads-run-every-frame.js

```diff
@@ -103,6 +104,100 @@ test('edge activated hat thread not added twice', t => {
     });
 });
 
+
+/**
+ * Duplicating a sprite should also track duplicated edge activated hat in
+ * runtime's _edgeActivatedHatValues map.
+ */
+test('edge activated hat should trigger for both sprites when sprite is duplicated', t => {
+
+    // Project that is similar to loudness-hat-block.sb2, but has code on the sprite so that
+    // the sprite can be duplicated
+    const projectWithSpriteUri = path.resolve(__dirname, '../fixtures/edge-triggered-hat.sb3');
+    const projectWithSprite = readFileToBuffer(projectWithSpriteUri);
+
+    const vm = new VirtualMachine();
```

Comment:
```js
    const vm = new VirtualMachine();
    vm.attachStorage(makeTestStorage());

    // Start VM, load project, and run
    t.doesNotThrow(() => {
        vm.clear();
        vm.setCompatibilityMode(false);
        vm.setTurboMode(false);

        vm.loadProject(project).then(() => {
```

Is used by a lot of tests. Would it make sense to write this chunk as a test utility like `makeTestStorage` is?

Reply:
I think I'll probably make an ergonomics issue for this instead of adding it to this PR as it will affect nearly all of the integration tests.

Reply:
Tracking in #1844